packages/mds-compliance-engine/engine/helpers.ts

```diff
@@ -54,23 +53,35 @@ export async function getAllInputs() {
   }, {})
 }
 
-export async function getProviderInputs(provider_id: string) {
-  const deviceRecords = await db.readDeviceIds(provider_id)
-  const deviceIdSubset = deviceRecords.map((record: { device_id: UUID; provider_id: UUID }) => record.device_id)
-  const devices = await cache.readDevices(deviceIdSubset)
-  // Get last event for each of these devices.
-  const events = await cache.readEvents(deviceIdSubset)
+/**
+ *
+ * @param provider_id Provider ID to query for
+ * @param timestamp Timestamp to query state up till
+ * @returns All events and devices for a given provider
+ */
+export async function getProviderInputs(provider_id: string, timestamp: number = now()) {
+  const { events } = await IngestServiceClient.getEventsUsingOptions({
+    provider_ids: [provider_id],
+    time_range: { start: timestamp - days(2), end: timestamp },
+    grouping_type: 'latest_per_vehicle'
```

Comment:
```suggestion
    grouping_type: 'latest_per_vehicle',
    order: { column: 'timestamp', direction: 'ASC' }
```
You mean like that?

Reply:
Aha, I forgot we had that! Yup, that'd probably do the trick, gonna try it now :)

Reply:
Fixed in 82b0215! Thanks @nessur